pkg/cloudprovider/providers/azure/azure_standard.go

```diff
@@ -408,14 +408,29 @@ func (as *availabilitySet) GetInstanceTypeByNodeName(name string) (string, error
 	return string(machine.HardwareProfile.VMSize), nil
 }
 
-// GetZoneByNodeName gets zone from instance view.
+// GetZoneByNodeName gets availability zone for the specified node. If the node is not running
+// with availability zone, then it returns fault domain.
 func (as *availabilitySet) GetZoneByNodeName(name string) (cloudprovider.Zone, error) {
 	vm, err := as.getVirtualMachine(types.NodeName(name))
```

Comment:
Why don't we use instance metadata for this as well similar to `vmss`?

Reply:
there're two ways actually,

- if cloud-controller-manager is used, then GetZoneByProviderID is used, which calls GetZoneByNodeName internally. Metadata won't work for this way because cloud-controller-manager is a master component.
- If kube-controller-manager is used, then GetZone() is used, which calls  instance metadata to fetch zone.